src/device_trezor/trezor/protocol.cpp

```diff
@@ -267,9 +349,28 @@ namespace tx {
     return std::string(buff, offset);
   }
 
+  ::crypto::secret_key compute_enc_key(const ::crypto::secret_key & private_view_key, const std::string & aux, const std::string & salt)
+  {
+    uint8_t hash[32];
+    KECCAK_CTX ctx;
+    ::crypto::secret_key res;
+
+    keccak_init(&ctx);
+    keccak_update(&ctx, (const uint8_t *) private_view_key.data, 32);
+    if (!aux.empty()){
+      keccak_update(&ctx, (const uint8_t *) aux.data(), aux.size());
+    }
+    keccak_finish(&ctx, hash);
+    keccak(hash, 32, hash, 32);
+
+    hmac_keccak_hash(hash, (const uint8_t *) salt.data(), salt.size(), hash, 32);
+    memcpy(res.data, hash, 32);
+    return res;
```

Comment:
Looks like hash is used as a secret key, so you may want to memwipe it.

Reply:
done